src/web-server/app.py

```diff
@@ -1,28 +1,53 @@
+import csv
 import json
 import os
+import random
 import sqlite3
+import urllib.parse
 from collections import namedtuple
 from contextlib import contextmanager
-from csv import reader
 from multiprocessing import Process, Queue
 
-from rauth import OAuth2Service, OAuth1Service
-
-from IGNORE_scheme_debug import Buffer, tokenize_lines, debug_eval, scheme_read
-
 import black
 import requests
-from flask import Flask, jsonify, make_response, redirect, request
+from english_words import english_words_set as words  # list of words to generate links
+from flask import Flask, jsonify, make_response, redirect, request, session, url_for
+from flask_oauthlib.client import OAuth
+from werkzeug import security
 
+from IGNORE_scheme_debug import Buffer, debug_eval, scheme_read, tokenize_lines
+from IGNORE_secrets import SECRET
 from formatter import scm_reformat
 
-CSV = "https://docs.google.com/spreadsheets/u/1/d/1v3N9fak7a-pf70zBhAIUuzplRw84NdLP5ptrhq_fKnI/export?format=csv&id=1-1v3N9fak7a-pf70zBhAIUuzplRw84NdLP5ptrhq_fKnI&gid=0"
+CSV = (
+    "https://docs.google.com/spreadsheets/u/1/d/1v3N9fak7a-pf70zBhAIUuzplRw84NdLP5ptrhq_fKnI/export?format=csv&id=1"
+    "-1v3N9fak7a-pf70zBhAIUuzplRw84NdLP5ptrhq_fKnI&gid=0 "
+)
+
+CONSUMER_KEY = "61a-web-repl"
+
+BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+STAFF_EMAIL_PATH = os.path.join(BASE_DIR, "staff_emails.csv")
+
+with open(STAFF_EMAIL_PATH) as f:
+    reader = csv.reader(f)
+    next(reader)
+    STAFF_EMAILS = set(x[0] for x in reader)
 
 app = Flask(__name__, static_url_path="", static_folder="")
+app.secret_key = SECRET
+
 ServerFile = namedtuple(
     "ServerFile", ["short_link", "full_name", "url", "data", "discoverable"]
 )
 
+NOT_FOUND = "NOT_FOUND"
```

Comment:
So I decided that returning `None`, an arbitrary hardcoded string, or a dict was too confusing - now, it either returns one of an enumerated series of strings, or returns a dict with the desired info.